.dev/compare_branches.R

```diff
@@ -318,17 +319,44 @@ run_workflow <- function(what, packages, linter_names, branch, number) {
 
   # safe to use force=TRUE because we're in temp_repo
   if (what == "pr") {
-    # pr_fetch doesn't expose this so use this to reset
     gert::git_branch_checkout("main", force = TRUE)
-    usethis::pr_fetch(number)
+    gert::git_checkout_pull_request(number)
   } else {
-    gert::git_branch_checkout(branch, force = TRUE)
+    # gert::git_branch_checkout() doesn't support checking out tags
+    #   directly (like CLI 'git checkout' flexibly does); use this workaround instead
+    tryCatch(
+      gert::git_branch_checkout(branch, force = TRUE),
+      error = function(cond) {
+        tag_hash <- gert::git_tag_list(branch)$commit
+        if (length(tag_hash) == 0L) {
+          stop("Unable to find branch or tag '", branch, "'")
+        }
+        # if an actual tag is provided, we'll get 0 or 1 results. but
+        #   git_tag_list() supports wildcards too, e.g. git_tag_list("v2*"), so safeguard:
+        if (length(tag_hash) > 1L) {
+          warning("Matched more than one tag! Selecting the first of: ", toString(tag_hash))
```

Comment:
Might be more helpful to specify the selected tag name rather than the associated commit hash.